src/test/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandlerQueueTest.java

```diff
@@ -0,0 +1,175 @@
+package com.sonymobile.tools.gerrit.gerritevents;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.sonymobile.tools.gerrit.gerritevents.dto.GerritEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.RefUpdated;
+
+//CS IGNORE MagicNumber FOR NEXT 600 LINES. REASON: Test data.
+
+/**
+ * Tests the GerritHandler queueing.
+ */
+public class GerritHandlerQueueTest {
+
+    private GerritHandler handler;
+    SlowEventListener listener;
+
+    /**
+     * Initialize.
+     */
+    @Before
+    public void setup() {
+        listener = new SlowEventListener();
+    }
+
+    /**
+     * Shuts down the GerritHandler.
+     */
+    @After
+    public void shutDown() {
+        if (handler != null) {
+            handler.shutdown(true);
+        }
+        handler = null;
+    }
+
+    /**
+     * Put given number of events in the queue.
+     *
+     * @param number
+     *            number of events
+     */
+    private void postEventsToQueue(int number) {
+        for (int i = 0; i < number; i++) {
+            handler.post(new RefUpdated());
+        }
+
+    }
+
+    /**
+     * Default number of executors is used.
+     *
+     * @throws Exception if occurred
+     */
+    @Test
+    public void testDefaultNumberofParallelWorkers() throws Exception {
+        handler = new GerritHandler();
+
+        listener = new SlowEventListener();
+        handler.addListener(listener);
+        postEventsToQueue(5);
+        waitForEventsProcessed();
+        assertThat(handler.getLargestPoolSize(), equalTo(3));
```

Comment:
Better to use the constant here, so an onlooker understands it tests the default number